app/templates/main/map.html

```diff
@@ -106,509 +111,630 @@
+              $("#done").css('width', trimmed + '%').attr('aria-valuenow', trimmed).text(progress + '%')
+              var chunk = evt.target.result;
+              oboeInstance.emit('data', chunk); // callback for handling read chunk
+            } else {
+              console.log("Read error: " + evt.target.error);
+              return;
+            }
+            if (offset >= fileSize) {
+              os.emit('done');
+              console.log("Done reading file");
+              endTime = Date.now();
+              $("#stats").text("Time taken: " + ((endTime - startTime) / 1000).toFixed(2) +
+                "s for file size " + (fileSize / (1024 * 1024)).toFixed(2) + " MB")
+              lv.ProcessView()
+              return;
+            }
+
+            // of to the next chunk
+            chunkReaderBlock(offset, chunkSize, file);
+          }
+
+          chunkReaderBlock = function (_offset, length, _file) {
+            var r = new FileReader();
+            var blob = _file.slice(_offset, length + _offset);
+            r.onload = readEventHandler;
+            r.readAsText(blob);
+          }
+
+          // now let's start the read with the first block
+          chunkReaderBlock(offset, chunkSize, file);
+        }
+
+
+        // INIT CSV
+        var csv = [
+          ["name of area", "end time", "duration"]
+        ];
+        weekStartMS = 0;
+        weekEndMS = 0;
+
+        // On click of submit button start csv generation process
+        $('#submitProcess').on('click', function (e) {
+          var errString = "";
+          $('#submitErrors').remove();
+          // if (!uploadURL) {
+          //   errString += "No file has been uploaded. Please select a file.";
+          // }
+          if (Object.keys(boxes).length <= 0) {
+            errString += " Please select at least one bounding box."
+          }
+          if (!$('#filename').val()) {
+            errString += " Please enter a name for your file."
+          }
+          if ($("#startDay").val() === "") {
+            errString += " Please select a valid start day"
+          }
+          if ($("#endDay").val() === "") {
+            errString += " Please select a valid end day."
+          }
+
+          if (errString.length > 0) {
+            $('#submitProcess').before("<div id='submitErrors' class='col-md-12'><div class='alert alert-danger'>" + errString + "<\/div><\/div>");
+          } else {
+            obj = processData(uploadURL, boxes, $('#email').val(), $('#startDay').val(), $('#endDay').val());
+            inside = new Array(obj.boxes.length).fill(false);
+            startTimes = new Array(obj.boxes.length).fill(null);
+            durations = new Array(obj.boxes.length).fill(0);
+            var newOboe = oboe();
+
+            newOboe.node('locations.*', function (location) {
+              if (weekEndMS === 0) {
+                weekEndMS = moment.unix(location.timestampMs / 1000).day(obj.dayEnd).hour(23).minute(59).unix() * 1000;
```

Comment:
minor nit but i think `moment(Number)` takes in ms it might be easier to use that instead of `moment.unix(Number)` and converting to seconds